app/interface/deps/services/course_deps.py

```
@@ -0,0 +1,11 @@
+from fastapi import Depends
+
+from app.application.services.course import AllCoursesDetailService
+from app.infrastructure.scrapers import LmsScraper
+from app.interface.deps import get_lms_scraper
+
+
+def get_all_courses_detail_service(
+    scraper: LmsScraper = Depends(get_lms_scraper),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)